### ISSUE TITLE
[BUG?] Does the alanine dipeptide test system have a mixture of combining rules?

### ISSUE THREAD

**Issue by lohedges** [OpenBioSim/sire#155]
I don't know if this is an issue or not, but it confused me. If I load our canonical alanine dipeptide test system then I get no warning from Sire. However, on writing to file and reading back in I now get a warning regarding the presence of both arithmetic _and_ geometric combining rules being present in the file. Assuming that we don't modify the existing information, why is this not the case first time around?

```python
In [1]: import sire as sr

In [2]: mols = sr.load_test_files("ala.top", "ala.crd")

In [3]: sr.save(mols, "test.prm7")
Out[3]: ['/home/lester/Code/openbiosim/sire/test.prm7']

In [4]: sr.save(mols, "test.rst7")
Out[4]: ['/home/lester/Code/openbiosim/sire/test.rst7']

In [5]: mols = sr.load(["test.prm7", "test.rst7"])

WARNINGS encountered when parsing the topology:
The LJ parameters in this Amber Parm file use both arithmetic and geometric combining rules. Sire will use arithmetic combining rules for all LJ parameters.====


Silence these warnings by passing `show_warnings=False` when loading.
```

**Comment by lohedges** [comment]
Ah, I see, `load_test_files` must have been silencing the warnings by default, hence why I never saw them.

**Comment by chryswoods** [comment]
I am investigating this separately as I don't think the file does contain a mixture (but my tests last night suggested it did). There is also weirdness that writing and then re-reading the file gives the same warning, despite writing definitely using arithmetic... I think we can keep this issue open while resolving whether this is a bug, and if so, then fixing it.

**Comment by chryswoods** [comment]
I've found and fixed the problem. The test to check if arithmetic rules weren't used was tripped accidentally for ghost atoms. This is because `0.5( sigma_a + sigma_b ) != 0` when `sigma_a` or `sigma_b` are zero, while the PRMTOP format records this as being zero. In contrast, for geometric rules, `sqrt(sigma_a * sigma_b)` is zero, hence why the code thought that geometric rules were used.

I've fixed it by adding a condition to check when both the PRMTOP and the combining rules say that epsilon_ij is zero, and to not then check further for exceptions or combining rules.

The impact of this bug is just the warning message. It didn't change the combining rules. It is more of an annoyance. I've nearly got `feature_force_lever` working so will merge it in with that.
